src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Queries/LoadProductsQueryHandler.cs

```diff
@@ -1,41 +1,207 @@
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using AutoMapper;
-using VirtoCommerce.ExperienceApiModule.Core;
+using MediatR;
+using VirtoCommerce.CoreModule.Core.Common;
+using VirtoCommerce.CoreModule.Core.Currency;
+using VirtoCommerce.ExperienceApiModule.Core.Extensions;
 using VirtoCommerce.ExperienceApiModule.Core.Index;
 using VirtoCommerce.ExperienceApiModule.Core.Infrastructure;
+using VirtoCommerce.ExperienceApiModule.Core.Models;
+using VirtoCommerce.MarketingModule.Core.Model.Promotions;
+using VirtoCommerce.Platform.Core.Common;
 using VirtoCommerce.SearchModule.Core.Model;
 using VirtoCommerce.SearchModule.Core.Services;
+using VirtoCommerce.XDigitalCatalog.Extensions;
+using VirtoCommerce.XPurchase;
+using VirtoCommerce.XPurchase.Commands;
 
-namespace VirtoCommerce.ExperienceApiModule.DigitalCatalog.Queries
+namespace VirtoCommerce.XDigitalCatalog.Queries
 {
     public class LoadProductsQueryHandler : IQueryHandler<LoadProductQuery, LoadProductResponse>
     {
+        private readonly IMediator _mediator;
+        private readonly ICartAggregateRepository _cartAggregateRepository;
         private readonly IMapper _mapper;
         private readonly ISearchProvider _searchProvider;
- 
-        public LoadProductsQueryHandler(ISearchProvider searchProvider, IMapper mapper)
+
+        public LoadProductsQueryHandler(ISearchProvider searchProvider, IMapper mapper, IMediator mediator, ICartAggregateRepository cartAggregateRepository)
         {
             _searchProvider = searchProvider;
             _mapper = mapper;
+            _mediator = mediator;
+            _cartAggregateRepository = cartAggregateRepository;
         }
 
         public virtual async Task<LoadProductResponse> Handle(LoadProductQuery request, CancellationToken cancellationToken)
         {
             var result = new LoadProductResponse();
+            var requestFields = new List<string>();
```

Comment:
I've detected a huge code duplication with `SearchProductQueryHandler`, could you leave only the one handler that will implement the two `IQueryHandler<LoadProductQuery, >` and `IQueryHandler<SearchProductQuery, >` interfaces instead?